pkg/asset/machines/ibmcloud/machines.go

```diff
@@ -120,6 +132,44 @@ func provider(clusterID string,
 	}, nil
 }
 
+func getDedicatedHostNameForZone(clusterID string, role string, dhosts []ibmcloud.DedicatedHost, zone string) (string, error) {
+	var defaultName string
+	switch role {
+	case "master":
+		defaultName = fmt.Sprintf("%s-dhost-control-plane-%s", clusterID, zone)
+	case "worker":
+		defaultName = fmt.Sprintf("%s-dhost-compute-%s", clusterID, zone)
+	default:
+		return "", fmt.Errorf("invalid machine role %v", role)
+	}
+
+	for _, dhost := range dhosts {
+		if dhost.Zone == zone {
+			return defaultName, nil
+		}
+	}
+
+	client, err := ibmcloudconfig.NewClient()
+	if err != nil {
+		return "", err
+	}
+
+	for _, dhost := range dhosts {
+		if dhost.ID != "" {
+			dh, err := client.GetDedicatedHostByID(context.TODO(), dhost.ID, strings.Join(strings.Split(zone, "-")[:2], "-"))
+			if err != nil {
+				return "", err
+			}
+
+			if *dh.Zone.Name == zone {
+				return *dh.Name, nil
```

Comment:
If the MAO is expecting the name of the dedicated host rather than the ID, why are we asking the user to provide the ID rather than the name?

Reply:
Just to make it a little more clear that the dedicated host already exists (not that installer will create a new dedicated host with that name). But I can update the MachinePool to use name, if that's better.

Reply:
I'm interested in consistency between the installer config and the MAO config. The installer can (and should) check to verify that the dedicated host with the name exists, so I don't see an issue with the user incorrectly running an installation expecting the dedicated host to be created with the specified name. I also don't think that the user would even have that expectation since it would be pretty unique behavior for the installer to take a name for a resource that the installer is going to create.

Reply:
Ok understood - will adjust